packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -120,6 +120,62 @@ lambdaAction.outputArtifact('Out2'); // returns the named output Artifact, or th
 See [the AWS documentation](https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html)
 on how to write a Lambda function invoked from CodePipeline.
 
+### Lambda deployed through CodePipeline
+
+If you want to deploy your Lambda through CodePipeline,
+you need to override the Parameters that are present in the Asset of the Lambda Code.
+Note that your Lambda must be in a different Stack than your Pipeline.
+The Lambda itself will be deployed, alongside the entire Stack it belongs to,
+using a CloudFormation CodePipeline Action. Example:
+
+```typescript
+const lambdaCode = lambda.Code.asset('path/to/directory/or/zip/file'); 
+const lambda = new lambda.Function(lambdaStack, 'Lambda', {
+  code: lambdaCode,
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NodeJS810,
+});
+// other resources that your Lambda needs, added to the lambdaStack...
+
+const pipeline = new codepipeline.Pipeline(pipelineStack, 'Pipeline');
+// add the source code repository containing this code to your Pipeline,
+// and the source code of the Lambda Function, if they're separate
+pipeline.addStage({
+  name: 'Source',
+  actions: [
+    // ...
+  ],
+});
+// add a build Action to your Pipeline, that calls `cdk synth` on the lambdaStack,
+// and saves it to some file, and a separate build for your Lambda source code - if needed
+pipeline.addStage({
+  name: 'Build',
+  actions: [
+    lambdaBuildAction,
+    cdkBuildAction,
+  ],
+});
+// finally, deploy your Lambda code
+const parameterOverrides: { [name: string]: any } = {};
+parameterOverrides[lambdaCode.asset.bucketNameParam.logicalId] = lambdaBuildAction.outputArtifact.bucketName;
+parameterOverrides[lambdaCode.asset.objectKeyParam.logicalId] = lambdaBuildAction.outputArtifact.objectKey;
+pipeline.addStage({
```

Comment:
Can you build the next layer on top of this as well?

So that we basically do:

```ts
const stack = new MyStack(app, 'MyStack');

stack.addToPipeline(pipeline);  // (*)
```

And make that work? What will we need to map assets used in the application to build artifacts? Can we even find a list of all assets, so that the `addToPipeline()` call can determine that we satisfied all requirements? Etc.

// (*) I know this exact method call will not work because the package dependency arrows will point the wrong way, but I'm mirroring the `build.addToPipeline()` etc. methods here.

Reply:
We are actually planning to remove all the addTpPipeline methods since we are moving all the pipeline integration classes to a centralized package. But maybe something like an integration class would work.

Reply:
But yeah, this is waaayy to verbose

Reply:
> Can you build the next layer on top of this as well?
>
> So that we basically do:
>
> ```ts
> const stack = new MyStack(app, 'MyStack');
>
> stack.addToPipeline(pipeline);  // (*)
> ```
> And make that work? What will we need to map assets used in the application to build artifacts? Can we even find a list of all assets, so that the addToPipeline() call can determine that we satisfied all requirements? Etc.
>
> // (*) I know this exact method call will not work because the package dependency arrows will point the wrong way, but I'm mirroring the build.addToPipeline() etc. methods here.

We can't be so high level, unfortunately. You wouldn't know which Pipeline Artifacts to use to override each Asset - the customer needs to specify this.

Reply:
I know, but what I'm saying is trying to get as close to that as possible.